### DISCUSSION TITLE
Global Policies

### DISCUSSION THREAD

**Discussion by karras** [dani-garcia/vaultwarden#3725] [Ideas]
I might have missed already closed or resolved discussions/issues regarding this topic, if that's the case I'm sorry.

At the moment it seems not possible to apply global policies (outside of an organization) to all users. That would be ideal for organizations which do not seek to provide an organization-wide password store but instead want to offer a personal password store service only.

In short could a feature request be supported (also financially) to allow the configuration of global/instance-wide policies instead of only having them bound to an organization?

This would simplify the onboarding process, and not require users to join an organization only for the sake of enforcing policies (e.g. master password strength).

**Comment by BlackDex** [comment]
Are there other policies besides password strength?

The main issue with some is that they are tight to an org. Without a user beloning to an org, those will not become active.

Modifying it at runtime by somehow letting the server modify the web-vault code is difficult too, since the web-vault can change which needs adjustments or even worse version checks which can be altered.

I haven't looked into the master password settings my self for a global item, so not sure if there is something to be done there.

And maybe the password generator.

**Reply by karras** [reply]
Ideally there would be at least the following policies (highest priorities first):

* Require everybody during the sign-up to configure MFA
  - AFAIK this was already reviewed in other discussions/issues and seems not possible without major changes in the onboarding process?
  - This would be extremely nice and avoid the need to frequently check if someone has disabled MFA
* Require a certain strength for the master password
* Disallow the creation of Vaultwarden organizations
  - Handy for use cases where a company only wants to provide an internal personal password manager but not use it for shared secrets
* Provide a preconfigured/enforced password generator strength
* Disable Send capabilities (already covered today in the env settings)

**Comment by BlackDex** [comment]
> Ideally there would be at least the following policies (highest priorities first):
> 
> * Require everybody during the sign-up to configure MFA
>   
>   * AFAIK this was already reviewed in other discussions/issues and seems not possible without major changes in the onboarding process?
>   * This would be extremely nice and avoid the need to frequently check if someone has disabled MFA

As you mentioned it your self, the onboarding process is not build for this. I can think of some workarounds, like a job in the background which gives the user a grace period for example and else disables the user. But that would cause issues i think. Aan other option would be pre-configure mail token.

> * Require a certain strength for the master password

This can be configured somewhere in the web-vault code and then build it your self ofcourse. There isn't an easy global way or runtime adjustment. Maybe a sed replace on the js code could be an option.

> * Disallow the creation of Vaultwarden organizations

This already is an option. You can configure a mail address who is allowed to create orgs, others are denied.

>   
>   * Handy for use cases where a company only wants to provide an internal personal password manager but not use it for shared secrets
> * Provide a preconfigured/enforced password generator strength

Same as for the master password strength. Maybe a sed replace on the js code can do something, else rebuild the web-vault your self.

> * Disable Send capabilities (already covered today in the env settings)

That is already an option indeed.

**Comment by andrepaulotek** [comment]
Would it be possible to automatically add all users (and new ones whey they create their account) to an organization?
This organization wouldn't have any login details, but it would allow to set these policies to everyone.
